solver/src/driver.rs

```diff
@@ -71,34 +73,56 @@ impl Driver {
             .context("failed to get uniswap pools")?;
         tracing::debug!("got {} AMMs", amms.len());
 
-        let liquidity = limit_orders
+        let liquidity: Vec<Liquidity> = limit_orders
             .into_iter()
             .map(Liquidity::Limit)
             .chain(amms.into_iter().map(Liquidity::Amm))
             .collect();
 
-        // TODO: order validity checks
-        // Decide what is handled by orderbook service and what by us.
-        // We likely want to at least mark orders we know we have settled so that we don't
-        // attempt to settle them again when they are still in the orderbook.
-        let settlement = match self.solver.solve(liquidity).await? {
-            None => return Ok(()),
-            Some(settlement) => settlement,
-        };
-        info!("Computed {:?}", settlement);
-        if settlement.trades.is_empty() {
-            info!("Skipping empty settlement");
-        } else {
-            // TODO: check if we need to approve spending to uniswap
-            settlement_submission::submit(
+        let mut settlements: Vec<(&Box<dyn Solver>, Settlement)> =
+            join_all(self.solver.iter().map(|solver| {
+                let liquidity = liquidity.clone();
+                async move { (solver, solver.solve(liquidity).await) }
+            }))
+            .await
+            .into_iter()
+            .filter_map(|(solver, settlement)| {
+                let settlement = settlement.ok()??;
+                info!(
+                    "{} found solution with objective value: {}",
+                    solver,
+                    settlement.objective_value()
+                );
+                Some((solver, settlement))
+            })
+            .collect();
+
+        // Sort by key yields ascending order, but we want to start with the highest objective value, so reverse iterator in loop
+        settlements.sort_by_key(|(_, settlement)| settlement.objective_value());
```

Comment:
You can use `std::cmp::Reverse` to wrap objective_value and avoid having to reverse the iterator